Fixed quit bug

bmgough opened the pull request:
The quit function was not terminating px.exe processes on my machine.  I tried send_signal(SIGTERM) instead of send_signal(CTRL_C_EVENT) but that didn't work either.  In the end I replaced send_signal(CTRL_C_EVENT) with kill().  Kill may be harsh but it was the only method I found that worked.

genotrance commented:
Does the --quit never work, or sometimes works and fails at other times? 

Presume you are running this on the PyInstaller version and not the Python script (we'll just kill any random Python.exe process in this case).

Also presume the --quit is being invoked in the same location (by the same executable in same location) as the one that is currently running, without which this condition won't ever be successful:-

`if p.exe() == sys.executable:`

Of course, if p.kill() is working, it means you are :)

Lastly, I had checked in code recently cause Px wouldn't respect CTRL-C in some cases.

`signal.signal(signal.SIGINT, signal.SIG_DFL)`

Without this, it wouldn't always exit as expected. I presume your code has this already included.

bmgough commented:
In response to “Does the --quit never work, or sometimes works and fails at other times?”

 

It never worked for me; however I only tested the --quit parameter a few times before I updated the code.

 

I used pyinstaller version 3.2.1 and python version 2.7.13 to build.  I tried to build with python 3.6.1 but I couldn’t get pyinstaller to work with it.

 

I left signal.signal(signal.SIGINT, signal.SIG_DFL) in the serve_forever function.  I updated build.bat and added p.kill().  I didn’t want to submit a pull request for the changes to build.bat therefore I created a new branch and cherry-picked the p.kill() commit.

 

If it matters, I tested using the following 2 build.bat files (I don’t have much experience with pyinstaller):

@echo off

del /q px.spec

rmdir /s /q build

rmdir /s /q __pycache__

rmdir /s /q dist

 

pyinstaller --clean --onefile -w -i px.ico px.py

copy px.ini dist\.

copy *.txt dist\.

 

del /q px.spec

rmdir /s /q build

rmdir /s /q __pycache__

 

AND

 

@echo off

del /q px.spec

rmdir /s /q build

rmdir /s /q __pycache__

rmdir /s /q dist

 

pyinstaller --clean --noupx -w -i px.ico px.py

copy px.ini dist\px\.

copy *.txt dist\px\.

 

del /q px.spec

rmdir /s /q build

rmdir /s /q __pycache__

 

From: Ganesh Viswanathan [mailto:notifications@github.com] 
Sent: Friday, April 21, 2017 8:23 AM
To: genotrance/px <px@noreply.github.com>
Cc: bmgough <brad.m.gough@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [genotrance/px] Fixed quit bug (#8)

 

Does the --quit never work, or sometimes works and fails at other times?

Presume you are running this on the PyInstaller version and not the Python script (we'll just kill any random Python.exe process in this case).

Also presume the --quit is being invoked in the same location (by the same executable in same location) as the one that is currently running, without which this condition won't ever be successful:-

if p.exe() == sys.executable:

Of course, if p.kill() is working, it means you are :)

Lastly, I had checked in code recently cause Px wouldn't respect CTRL-C in some cases.

signal.signal(signal.SIGINT, signal.SIG_DFL)

Without this, it wouldn't always exit as expected. I presume your code has this already included.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/genotrance/px/pull/8#issuecomment-296266560> , or mute the thread <https://github.com/notifications/unsubscribe-auth/ANrW-L4mouufFuw21JosD7nIDnD1RHcnks5ryPQJgaJpZM4NEFrP> .  <https://github.com/notifications/beacon/ANrW-HRU4xwbhQl6DBrHroz6Poo1P_Hsks5ryPQJgaJpZM4NEFrP.gif>

genotrance commented:
Closing this PR - now that we override CTRL-C, it works much better than it did before.
